python/lsst/resources/s3.py

```diff
@@ -118,6 +125,25 @@ def _translate_client_error(err: ClientError) -> None:
 class S3ResourcePath(ResourcePath):
     """S3 URI resource path implementation class."""
```

Comment:
Please document the environment variable here in a notes section like we do in `HttpResourcePath`.